src/contextualbar/IContextualBarButton.ts

```diff
@@ -0,0 +1,28 @@
+import { RoomType } from '../rooms/RoomType';
+
+export interface IContextualBarButton {
+    /** A unique identifier for your button.  */
+    id: string;
+
+    /** The i18n title for your button. */
+    i18nTitle: string;
+
+    /**
+     * The icon class for your button.
+     * See: https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-theme/client/vendor/fontello/css/fontello.css
+     */
+    icon: string;
```

Comment:
I think this is a decent route to go, except it should be for the button's icon only.